src/main/java/duke/UserInterface.java

```diff
@@ -0,0 +1,533 @@
+                    prepositions);
+        }
+        return null;
+    }
+
+    /**
+     * Validates the input of the task to check for user error.
+     *
+     * @param taskType  the type of task (todo, deadline etc).
+     * @param taskParts the list of components that form the task.
+     * @throws DukeException error message if task is invalid
+     */
+    private void validateTask(String taskType, ArrayList<StringBuilder> taskParts) throws DukeException {
+        if (taskParts.isEmpty()) {
+            throw new DukeException("The description of a " + taskType + " cannot be empty.");
+        }
+        String prepositions = String.valueOf(taskParts.get(taskParts.size() - 1));
+        switch (taskType) {
+        case COMMAND_TODO:
+            if (!prepositions.isEmpty()) {
+                throw new DukeException("Todo tasks cannot include the preposition " + prepositions +" in the description.");
+            }
+            break;
+        case COMMAND_DEADLINE:
+            if (taskParts.get(0).toString().equals("")
+                    || taskParts.get(1).toString().equals("")) {
+                throw new DukeException("The description or deadline is incomplete.");
+            }
+            break;
+        case COMMAND_EVENT:
+            if (taskParts.get(0).toString().equals("") || taskParts.get(1).toString().equals("")) {
+                throw new DukeException("The description or time of event is incomplete.");
+            }
+        }
+    }
+
+
+    /**
+     * Splits an ArrayList<String> into an ArrayList<StringBuilder>, where each element
+     * can be used to construct a subclass of Task. Use this in conjunction with buildTask to construct a
+     * subclass of Task.
+     *
+     * @param input ArrayList of Strings that represent the user's input split by whitespace.
+     * @return an ArrayList<StringBuilder> containing the input split by the '/' character.
+     * Each element represents a String used to construct the task.
+     * The last element contains all prepositions used in a single string.
+     * e.g.: deadline return book /by Sunday /at library returns the following Arraylist:
+     * {"return book", "Sunday", "Library", "/by/at"}
+     * returns an empty list if input contains only a command.
+     */
+    private ArrayList<StringBuilder> splitTask(ArrayList<String> input) {
+        /* taskPartsIterator increments when a String containing '/' is encountered.
+         * inputIterator increments after each String in input is read.
+         * The loop ends when inputIterator exceeds the size of the ArrayList, indicating that all
+         * Strings in input have been read.
+         */
+        ArrayList<StringBuilder> taskParts = new ArrayList<>();
+        if (input.size() <= 1) {
+            return taskParts;
+        }
+        StringBuilder prepositions = new StringBuilder();
+        for (int taskPartsIterator = 0, inputIterator = 0; inputIterator < input.size(); ++taskPartsIterator) {
```

Comment:
Consider using iterator variable` i` instead which is found in the [coding standard conventions](https://se-education.org/guides/conventions/java/basic.html#naming).